### PR TITLE
Expand test coverage

### PR DESCRIPTION
**Pull request by joshcooper**
Expand test coverage to include Puppet6. Update README (see MODULES-8182) and project metadata.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -13,17 +13,20 @@ script:
...
+    - BEAKER_PUPPET_COLLECTION=puppet6 PUPPET_GEM_VERSION="~> 6.0"
 matrix:
   fast_finish: true
   include:
     -
       env: CHECK="syntax lint metadata_lint check:symlinks check:git_ignore check:dot_underscore check:test_file rubocop"
     -
       env: CHECK=parallel_spec
+    -
+      env: PUPPET_GEM_VERSION="~> 5.0" CHECK=parallel_spec
```
Comment (melissa) [inline .travis.yml]: Should this also have ` BEAKER_PUPPET_COLLECTION=puppet5`?

Comment (joshcooper) [reply .travis.yml]: That's a good question... I believe `BEAKER_PUPPET_COLLECTION` isn't used unless we're running beaker tests (as it seems to install/test against the right puppet gem)...

Comment (melissa) [reply .travis.yml]: Should we remove `BEAKER_PUPPET_COLLECTION` then to remove this bit of confusion?

Comment (melissa) [reply .travis.yml]: As a reference, maillist doesn't have this, and travis executes successfully with puppet 4, puppet5, and puppet 6
https://github.com/puppetlabs/puppetlabs-maillist_core/pull/8

Comment (joshcooper) [reply .travis.yml]: ah yeah, let me update this PR, and if everything still passes I'll do the others

**Pull request merged**